static/js/copyPasteEvents.js

```diff
@@ -130,12 +242,84 @@ var htmlDecode = function(input) {
+// here we find the comment id on a position [line, column]. This function is used to get the comment id
+// of one line when there is ONLY text selected. E.g In the line with comment, <span class='comment...'>something</span>,
+// and user copies the text 'omethin'. The span tags are not copied only the text. So as the comment is
+// applied on the selection we get the commentId using the first position selected of the line.
+// P.S: It's not possible to have two or more comments when there is only text selected, because for each comment
+// created it's generated a <span> and to copy only the text it MUST NOT HAVE any tag on the selection
+exports.getCommentIdOnFirstPositionSelected = function() {
+  var attributeManager = this.documentAttributeManager;
+  var rep = this.rep;
+  var commentId = _.object(attributeManager.getAttributesOnPosition(rep.selStart[0], rep.selStart[1])).comment;
+  return commentId;
+};
+
+exports.hasCommentOnSelection = function() {
+  var hasComment;
   var attributeManager = this.documentAttributeManager;
   var rep = this.rep;
-  var selStartAttrib = _.object(attributeManager.getAttributesOnPosition(rep.selStart[0], rep.selStart[1])).comment;
-  var selEndAttrib = _.object(attributeManager.getAttributesOnPosition(rep.selEnd[0], rep.selEnd[1] - 1)).comment;
-  return selStartAttrib === selEndAttrib ? selStartAttrib : null;
+  var firstLineOfSelection = rep.selStart[0];
+  var firstColumn = rep.selStart[1];
+  var lastColumn = rep.selEnd[1];
+  var lastLineOfSelection = rep.selEnd[0];
+  var selectionOfMultipleLine = hasMultipleLineSelected(firstLineOfSelection, lastLineOfSelection);
+
+  if(selectionOfMultipleLine){
+    hasComment = hasCommentOnMultipleLineSelection(firstLineOfSelection,lastLineOfSelection, rep, attributeManager);
+  }else{
+    hasComment = hasCommentOnLine(firstLineOfSelection, firstColumn, lastColumn, attributeManager)
+  }
+  return hasComment;
+};
+
+var hasCommentOnMultipleLineSelection = function(firstLineOfSelection, lastLineOfSelection, rep, attributeManager){
+  var line = firstLineOfSelection;
+  var multipleLineSelectionHasComment = false;
+  for (line; line <= lastLineOfSelection; line++) {
+    var firstColumn = getFirstColumnOfSelection(line, rep, firstLineOfSelection);
+    var lastColumn = getLastColumnOfSelection(line, rep, lastLineOfSelection);
+    var hasComment = hasCommentOnLine(line, firstColumn, lastColumn, attributeManager)
+    if (hasComment){
+      multipleLineSelectionHasComment = true;
+    }
+  }
+  return multipleLineSelectionHasComment;
+};
+
+var getFirstColumnOfSelection = function(line, rep, firstLineOfSelection){
+  return line !== firstLineOfSelection ? 0 : rep.selStart[1];
+};
+
+var getLastColumnOfSelection = function(line, rep, lastLineOfSelection){
+  var lineLength = getLength(line, rep);
+  var positionOfLastCharacterSelected = rep.selEnd[1] - 1;
+  return line !== lastLineOfSelection ? lineLength : positionOfLastCharacterSelected;
+};
```

Comment:
suggestion: use `if/else` instead of ternary operator, so you won't have to calculate both `lineLength` and `positionOfLastCharacterSelected` for every line you need to get last selected column -- and you would only use one of these values anyway